File: frontend/lib/types/study.ts
// Study-journey category cards — mirror backend app/schemas/progress.py
// (CategoryCardOut / StudyOverviewOut).

export type JourneyState = "not_started" | "learning" | "reviewing" | "mastered";
export type MasteryLevel =
  | "not_started"
  | "needs_review"
  | "developing"
  | "proficient"
  | "mastered";

export interface CategoryCardOut {
  topic_id: string;
  topic_name: string;
  level: number;
  exam_type: string | null;
  total_questions: number;
  answered: number;
  new_count: number;
  due_count: number;
  mastery_level: MasteryLevel | string;
  topic_theta: number | null;
  accuracy_rate: number | null;
  journey_state: JourneyState | string;
  recommended_mode: "adaptive_practice" | "review" | null;
}

export interface StudyOverviewOut {
  categories: CategoryCardOut[];
}

File: ios/Rono/Models/Study.swift
import Foundation

enum JourneyState: String, Codable, Sendable {
    case notStarted = "not_started"
    case learning
    case reviewing
    case mastered
}

enum MasteryLevel: String, Codable, Sendable {
    case notStarted = "not_started"
    case needsReview = "needs_review"
    case developing
    case proficient
    case mastered
}

struct CategoryCardOut: Codable, Identifiable, Sendable, Hashable {
    let topicId: String
    let topicName: String
    let level: Int
    let examType: String?
    let totalQuestions: Int
    let answered: Int
    let newCount: Int
    let dueCount: Int
    let masteryLevel: String
    let topicTheta: Double?
    let accuracyRate: Double?
    let journeyState: String
    let recommendedMode: String?   // adaptive_practice | review | null

    var id: String { topicId }
    var mastery: MasteryLevel? { MasteryLevel(rawValue: masteryLevel) }
    var journey: JourneyState? { JourneyState(rawValue: journeyState) }
}

struct StudyOverviewOut: Codable, Sendable, Hashable {
    let categories: [CategoryCardOut]
}
